torch/testing/_asserts.py

```diff
@@ -157,8 +246,26 @@ def _check_attributes_equal(
     if check_dtype and actual.dtype != expected.dtype:
         return AssertionError(msg_fmtstr.format("dtype", actual.dtype, expected.dtype))
 
-    if check_stride and actual.stride() != expected.stride():
-        return AssertionError(msg_fmtstr.format("stride()", actual.stride(), expected.stride()))
+    actual_format = _TensorFormat(actual)
+    expected_format = _TensorFormat(expected)
+    if actual_format.is_sparse and expected_format.is_sparse:
+        if actual_format.sparse_format != expected_format.sparse_format:
+            return AssertionError(
+                f"Sparse format does not match: {actual_format.sparse_format} != {expected_format.sparse_format}"
+            )
+
+        if (
+            actual_format.sparse_format == "COO"
+            and check_is_coalesced
+            and actual.is_coalesced() != expected.is_coalesced()
+        ):
+            return AssertionError(msg_fmtstr.format("is_coalesced()", actual.is_coalesced(), expected.is_coalesced()))
+    elif not actual_format.is_sparse and not expected_format.is_sparse:
```

Comment:
If the `actual` and the `expected` format are unknown (and hence, incorrectly to be assumed as dense, see above), this logic will give always a false-positive result in case a new sparse format is introduced. In fact, it will be silent so that `actual_format == expected_format` check is never reached.